jni/getroot.c

```diff
@@ -71,6 +71,9 @@ int modify_task_cred_uc(struct thread_info* __kernel info)
 		return 1;
 
 	tsp = malloc(sizeof(*tsp));
+	if (!tsp)
+		return 1;
```

Comment:
The check is like this
```
if(modify_task_cred_uc(ti))
		goto end;
```
This translates to
```
if return value is true
		goto end;
```

The language C does not have a special type for boolean like for example C#. In C, anything which is not 0 is true.

Therefore we do not need to adapt the retval check.

Anyway I think it's better to use specific error codes like you did :)
I probably just returned 1 for all errors because I was lazy.